smart-contracts/assembly/contracts/NFT/NFT-internals.ts

```diff
@@ -1,282 +1,239 @@
-  nft1_ownerOf,
-  ownerKey,
-  ownerTokenKey,
-  symbolKey,
-  totalSupplyKey,
-} from './NFT';
-
-/**
- * Initialize all the properties of the NFT (contract Owner, counter to 0...)
- *
- * @param args - Args object serialized as a string containing:
- * - the token name (string)
- * - the token symbol (string).
- * - the totalSupply (u64)
- * - the baseURI (string)
- */
-export function _constructor(args: Args): void {
-  // This line is important. It ensures that this function can't be called in the future.
-  // If you remove this check, someone could call your constructor function and reset your smart contract.
-  assert(Context.isDeployingContract());
+import { u256 } from 'as-bignum/assembly';
 
-  const name = args.nextString().expect('name argument is missing or invalid');
-  const symbol = args
-    .nextString()
-    .expect('symbol argument is missing or invalid');
-  const totalSupply = args
-    .nextU64()
-    .expect('totalSupply argument is missing or invalid');
-  const baseURI = args
-    .nextString()
-    .expect('baseURI argument is missing or invalid');
+export const NAME_KEY: StaticArray<u8> = [0x01];
+export const SYMBOL_KEY: StaticArray<u8> = [0x02];
 
-  Storage.set(nameKey, name);
-  Storage.set(symbolKey, symbol);
-  Storage.set(totalSupplyKey, u64ToBytes(totalSupply));
-  Storage.set(baseURIKey, baseURI);
-  Storage.set(ownerKey, Context.caller().toString());
-  Storage.set(counterKey, u64ToBytes(initCounter));
-}
+export const BALANCE_KEY_PREFIX: StaticArray<u8> = [0x03];
+export const OWNER_KEY_PREFIX: StaticArray<u8> = [0x04];
+export const ALLOWANCE_KEY_PREFIX: StaticArray<u8> = [0x05];
+export const OPERATOR_ALLOWANCE_KEY_PREFIX: StaticArray<u8> = [0x06];
 
 /**
- * Increment the NFT counter
+ * Constructs a new NFT contract.
+ * @param binaryArgs - the binary arguments name and symbol
  */
-export function _increment(): u64 {
-  const currentID = bytesToU64(Storage.get(counterKey));
-  const newID = SafeMath.add(currentID, 1);
-  Storage.set(counterKey, u64ToBytes(newID));
-  return newID;
-}
-
-export function _updateBalanceOf(address: string, increment: boolean): void {
-  const balanceKey = stringToBytes('balanceOf_' + address);
-  const number = 1;
-
-  if (Storage.has(balanceKey)) {
-    const balance = bytesToU64(Storage.get(balanceKey));
-    let newBalance: u64;
-
-    if (increment) {
-      newBalance = SafeMath.add(balance, number);
-    } else {
-      newBalance = SafeMath.sub(balance, number);
-    }
-
-    Storage.set(balanceKey, u64ToBytes(newBalance));
-  } else {
-    assert(increment, 'Balance cannot be negative');
-    Storage.set(balanceKey, u64ToBytes(1));
-  }
-}
-
-export function _getBalanceOf(address: string): u64 {
-  const balanceKey = stringToBytes('balanceOf_' + address);
-
-  if (!Storage.has(balanceKey)) return 0;
-
-  return bytesToU64(Storage.get(balanceKey));
+export function _constructor(name: string, symbol: string): void {
+  Storage.set(NAME_KEY, stringToBytes(name));
```

Comment:
Why did you removed the `assert(Context.isDeployingContract());`? I have the feeling that now we can change the symbol and rename the the nft as we want.

Reply:
My idea is that this is not the final constructor but rather an helper that should be wrapped into the actual constructor. Thus, the `assert(Context.isDeployingContract());` isn't mandatory as it should be included in the final constructor